src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -57,6 +57,7 @@
     private AtomicBoolean rerouting = new AtomicBoolean();
     private volatile long registeredNextDelaySetting = Long.MAX_VALUE;
     private volatile ScheduledFuture registeredNextDelayFuture;
+    private volatile long lastAllocateUnassignedShardsRun = System.currentTimeMillis();
```

Comment:
Would it be useful to have 0 have the special "I've not even tried yet" meaning? I think setting it to now is kind-of funky because you aren't actually doing an allocation when you build the object.